.github/workflows/docker_publish.yaml

```diff
@@ -38,11 +39,16 @@ jobs:
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build frontend
+        run: |
+          yarn --cwd ./frontend install
+          yarn build:frontend
       - name: Build tag and push image to Docker hub
         uses: docker/build-push-action@v3
         with:
+          context: .
           platforms: linux/amd64,linux/arm64
-          push: true
+          push: ${{ github.event_name != 'workflow_dispatch' }}
```

Comment:
This enables us to run tests on this workflow without pushing the image

Reply:
Nice!